public/app/features/alerting/unified/components/settings/AlertmanagerConfig.tsx

```diff
@@ -0,0 +1,185 @@
+}
+
+export default function AlertmanagerConfig({ alertmanagerName, onDismiss, onSave, onReset }: Props): JSX.Element {
+  const { loading: isDeleting, error: deletingError } = useUnifiedAlertingSelector((state) => state.deleteAMConfig);
+  const { loading: isSaving, error: savingError } = useUnifiedAlertingSelector((state) => state.saveAMConfig);
+  const [showResetConfirmation, setShowResetConfirmation] = useState(false);
+
+  const readOnly = alertmanagerName ? isVanillaPrometheusAlertManagerDataSource(alertmanagerName) : false;
+  const styles = useStyles2(getStyles);
+
+  const {
+    currentData: config,
+    error: loadingError,
+    isSuccess: isLoadingSuccessful,
+    isLoading: isLoadingConfig,
+  } = useAlertmanagerConfig(alertmanagerName);
+
+  const defaultValues = {
+    configJSON: config ? JSON.stringify(config, null, 2) : '',
+  };
+
+  const {
+    register,
+    setValue,
+    setError,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues,
+  });
+
+  useEffect(() => {
+    if (savingError) {
+      setError('configJSON', { type: 'deps', message: savingError.message });
+    }
+  }, [savingError, setError]);
+
+  useEffect(() => {
+    if (deletingError) {
+      setError('configJSON', { type: 'deps', message: deletingError.message });
+    }
+  }, [deletingError, setError]);
+
+  // manually register the config field with validation
+  register('configJSON', {
+    required: { value: true, message: 'Configuration cannot be empty' },
+    validate: (value: string) => {
+      try {
+        JSON.parse(value);
+        return true;
+      } catch (e) {
+        return e instanceof Error ? e.message : 'JSON is invalid';
+      }
+    },
+  });
+
+  const handleSave = handleSubmit(
+    (values: FormValues) => {
+      onSave(alertmanagerName, defaultValues.configJSON, values.configJSON);
+    },
+    (errors) => {
+      console.error(errors);
+    }
+  );
+
+  const isOperating = isLoadingConfig || isDeleting || isSaving;
+
+  /* loading error, if this fails don't bother rendering the form */
+  if (loadingError) {
+    return (
+      <Alert severity="error" title="Failed to load Alertmanager configuration">
+        {loadingError.message ?? 'An unkown error occurred.'}
+      </Alert>
+    );
+  }
+
+  /* resetting configuration state */
+  if (isDeleting) {
+    return (
+      <Alert severity="info" title="Resetting Alertmanager configuration">
+        It might take a while...
+      </Alert>
+    );
+  }
+
+  return (
+    <div className={styles.container}>
+      {/* form error state */}
+      {errors.configJSON && (
+        <Alert severity="error" title="Oops, something went wrong">
+          {errors.configJSON.message || 'An unknown error occurred.'}
+        </Alert>
+      )}
+
+      {isLoadingSuccessful && (
+        <div className={styles.content}>
+          <AutoSizer>
+            {({ height, width }) => (
+              <>
```

Comment:
nit: unneeded fragment here (there's an eslint rule for it that can catch/auto fix)

Reply:
nice catch!